site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,742 @@
+
+### Why Do You Need This Plugin?
+
+This plugin can assist you with your internationalization.
+It is based on [Fluent](https://projectfluent.org/)—an excellent localization system built by [Mozilla](https://mozilla.org/).
+This system has a very powerful and elegant syntax that lets you write natural-sounding translations efficiently.
+
+In essence, you can extract the things that should adjust based on the user's locale to some text files that you put next to your code.
+You can then use this plugin to load these localizations.
+The plugin will automatically determine the user's locale and make your bot speak the correct language.
+
+In the following, we will call these text files _translation files_.
+They need to follow the syntax of Fluent.
 
-The basic thing you need are the translations.
-Normally they are stored in your bot structure as yaml files (but this plugin supports other ways to handle them.)
+## Getting Started
+
+> This section describes setting up your project structure and where to put your translation files.
+> If you are familiar with this, [skip ahead](#usage) to see how to install and use the plugin.
+
+There are [multiple ways](#adding-translations) to add more languages to your bot.
+The easiest way is to create a folder with your Fluent translation files.
+Usually, the name of that folder is going to be `locales/`.
+The translation files should have the extension `.ftl` (fluent).
+
+Here is an example project structure:
+
+```asciiart:no-line-numbers
+.
+├── bot.ts
+└── locales/
+    ├── de.ftl
+    ├── en.ftl
+    ├── it.ftl
+    └── ru.ftl
+```
+
+If you're unfamiliar with the Fluent file (`.ftl`) syntax, you can have a look at their syntax guide here: <https://projectfluent.org/fluent/guide>
+
+Here is an example translation file for English, called `locales/en.ftl`:
+
+```fluent
+start = Hi, how can I /help you?
+help = 
+    Send me some text, and I can make it bold for you.
+    You can change my language using the /language command.
+```
+
+The German equivalent would be called `locales/de.ftl` and look like this:
+
+```fluent
+start = Hallo, wie kann ich dir helfen? /help
+help = 
+    Schick eine Textnachricht, die ich für dich fett schreiben soll.
+    Du kannst mit dem Befehl /language die Spache ändern.
+```
+
+In your bot, you can now use these translations through this plugin.
+It will make them available through `ctx.t`:
+
+```ts
+bot.command("start", async (ctx) => {
+  await ctx.reply(ctx.t("start"));
+});
 
-```plaintext
-yaml and json are ok
-Example directory structure:
-├── locales
-│   ├── en.yaml
-│   ├── en-US.yaml
-│   ├── it.json
-│   └── ru.yaml
-└── bot.js
+bot.command("help", async (ctx) => {
+  await ctx.reply(ctx.t("help"));
+});
 ```
 
-These files contain translations like this:
+Whenever you call `ctx.t`, the locale of the current context object `ctx` is used to find the proper translation.
+Finding the proper translation is done using a function called _locale negotiator_.
```

Comment:
Can a function be called that?

Reply:
Idk. But can a function be called "language negotiator"?

Reply:
I can't today

Reply:
Np. Have fun.

Reply:
It should be called `localeNegotiator` anyway. It negotiates the locale, not the language. That's also what the docs explain btw. @dcdunkan do you want to fix this?

Reply:
It is already called `localeNegotiator` actually. I don't know if someone changed it while I was unavailable.